pkg/build/builder/docker.go

```diff
@@ -87,6 +87,29 @@ func (d *DockerBuilder) Build() error {
 	}
 
 	buildTag := randomBuildTag(d.build.Namespace, d.build.Name)
+	imageName := d.getDockerfileFrom(buildDir)
+	if imageName == "" {
+		return fmt.Errorf("Builder image id empty.")
+	}
+	var imageExists bool = true
+	_, err = d.dockerClient.InspectImage(imageName)
+	if err != nil {
```

Comment:
if the error is other than docker.ErrNoSuchImage, we should return it because it likely means that something is wrong with the Docker daemon.